lighthouse-core/lib/dependency-graph/base-node.js

```diff
@@ -30,10 +30,10 @@ class BaseNode {
   constructor(id) {
     this._id = id;
     this._isMainDocument = false;
-    /** @type {Node[]} */
-    this._dependents = [];
-    /** @type {Node[]} */
-    this._dependencies = [];
+    /** @type {Set<Node>} */
```

Comment:
I initially made this change since I've added quite a few `addDependency`, `removeDependency`, etc. calls. I can move this to a separate PR if you want since it probably doesn't make a large performance difference. (I expect the arrays are small enough for any impact to be negligible.)